amazon-kinesis-client/src/main/java/software/amazon/kinesis/common/StreamIdentifier.java

```diff
@@ -0,0 +1,69 @@
+package software.amazon.kinesis.common;
+
+import com.google.common.base.Joiner;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.experimental.Accessors;
+import software.amazon.awssdk.utils.Validate;
+
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+@EqualsAndHashCode @Getter @Accessors(fluent = true)
+public class StreamIdentifier {
+    private final Optional<String> accountIdOptional;
+    private final String streamName;
+    private final Optional<Long> streamCreationEpochOptional;
+
+    private static final String DELIMITER = ":";
+    private static final Pattern PATTERN = Pattern.compile(".*" + ":" + ".*" + ":" + "[0-9]*");
+
+    private StreamIdentifier(Optional<String> accountIdOptional, String streamName,
```

Comment:
It looks better if we have two constructor with all field as mandatory values

Reply:
Agree. Will change